test/unit/ServerConfigParams.test.ts

```diff
@@ -1,122 +1,156 @@
-        ).to.throw(ERROR_FIXED_USD_FEE_NEGATIVE);
+        expect(() => verifyServerConfiguration()).to.throw(
+          ERROR_FIXED_USD_FEE_NEGATIVE
+        );
       });
 
       it('Should pass even if all fee parameters are undefined', function () {
-        expect(() =>
-          verifyServerConfiguration({ app, contracts } as ServerConfigParams)
-        ).not.to.throw();
+        config.util.extendDeep(config, {
+          app: {
+            fixedUsdFee: undefined,
+            gasFeePercentage: undefined,
+            transferFeePercentage: undefined,
+          },
+        });
+        expect(() => verifyServerConfiguration()).not.to.throw();
       });
 
       it('Should pass even if all fee parameters are disabled', function () {
-        app.fixedUsdFee = 0;
-        app.transferFeePercentage = -1;
-        app.gasFeePercentage = 0;
-
-        expect(() =>
-          verifyServerConfiguration({ app, contracts } as ServerConfigParams)
-        ).not.to.throw();
+        config.util.extendDeep(config, {
+          app: {
+            fixedUsdFee: 0,
+            gasFeePercentage: 0,
+            transferFeePercentage: -1,
+          },
+        });
+
+        expect(() => verifyServerConfiguration()).not.to.throw();
       });
 
       it('Should pass if only gasFeePercentage is properly configured', function () {
-        app.gasFeePercentage = 0.1;
+        config.util.extendDeep(config, { app: { gasFeePercentage: 0.1 } });
 
-        expect(() =>
-          verifyServerConfiguration({ app, contracts } as ServerConfigParams)
-        ).not.to.throw();
+        expect(() => verifyServerConfiguration()).not.to.throw();
       });
 
       it('Should pass if only transferFeePercentage is properly configured', function () {
-        app.transferFeePercentage = 0.01;
+        config.util.extendDeep(config, {
+          app: { transferFeePercentage: 0.01 },
+        });
 
-        expect(() =>
-          verifyServerConfiguration({ app, contracts } as ServerConfigParams)
-        ).not.to.throw();
+        expect(() => verifyServerConfiguration()).not.to.throw();
       });
 
       it('Should pass if only fixedUsdFee is properly configured', function () {
-        app.fixedUsdFee = 1;
+        config.util.extendDeep(config, { app: { fixedUsdFee: 1 } });
 
-        expect(() =>
-          verifyServerConfiguration({ app, contracts } as ServerConfigParams)
-        ).not.to.throw();
+        expect(() => verifyServerConfiguration()).not.to.throw();
       });
 
       it('Should pass if more than one fee is properly configured', function () {
-        app.fixedUsdFee = 1;
-        app.gasFeePercentage = 0.1;
+        config.util.extendDeep(config, {
+          app: { fixedUsdFee: 1, gasFeePercentage: 0.1 },
+        });
+
+        expect(() => verifyServerConfiguration()).not.to.throw();
+      });
+    });
+
+    it('Should fail if the string representation of a BigNumber is incorrect', function () {
```

Comment:
Can I use a BigNumber? instead of the representation?

Reply:
This value comes from the config file. There we are using only simple types like strings and numbers. I don't think we will create a parameter as object (BigNumber) in that file.